plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -707,4 +749,136 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
+        if (codeTransformationSession?.getHilDownloadArtifact() != null) {
+            return codeTransformationSession?.getHilDownloadArtifact()
+        }
+
+        val jobId = codeTransformationSession?.getActiveJobId() as JobId
+        val downloadArtifactId = codeTransformationSession?.getHilDownloadArtifactId() as String
+
+        val tmpDir = try {
+            createTempDirectory("", null)
+        } catch (e: Exception) {
+            val errorMessage = "Unexpected error when creating tmp dir for HIL: ${e.localizedMessage}"
+            LOG.error { errorMessage }
+            throw e
+        }
+        try {
+            val hilArtifact = artifactHandler.downloadHilArtifact(jobId, downloadArtifactId, tmpDir)
+            if (hilArtifact != null) {
+                codeTransformationSession?.setHilTempDirectoryPath(tmpDir.toPath())
+                codeTransformationSession?.setHilDownloadArtifact(hilArtifact)
+            }
+            return hilArtifact
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun createDependencyReport(hilDownloadArtifact: CodeTransformHilDownloadArtifact): MavenDependencyReportCommandsResult {
+        val tmpDirPath = codeTransformationSession?.getHilTempDirectoryPath() as Path
+
+        try {
+            val copyPomForDependencyReport = createFileCopy(hilDownloadArtifact.pomFile, getPathToHilDependencyReportDir(tmpDirPath).resolve(HIL_POM_FILE_NAME))
+            setDependencyVersionInPom(copyPomForDependencyReport, hilDownloadArtifact.manifest.sourcePomVersion)
+        } catch (e: Exception) {
+            val errorMessage = "Unexpected error when preparing for HIL dependency report: ${e.localizedMessage}"
+            telemetry.error(errorMessage)
+            LOG.error { errorMessage }
+            return MavenDependencyReportCommandsResult.Failure
+        }
+
+        return codeTransformationSession?.createHilDependencyReportUsingMaven() as MavenDependencyReportCommandsResult
+    }
+
+    fun findAvailableVersionForDependency(pomGroupId: String, pomArtifactId: String): Dependency? {
+        try {
+            val report = parseXmlDependenciesReport(
+                getPathToHilDependencyReport(codeTransformationSession?.getHilTempDirectoryPath() as Path)
+            )
+            return report.dependencies?.first {
+                it.groupId == pomGroupId &&
+                    it.artifactId == pomArtifactId
+            }
+        } catch (e: NoSuchElementException) {
+            return null
+        } catch (e: Exception) {
+            val errorMessage = "Unexpected error when parsing HIL dependency report: ${e.localizedMessage}"
+            telemetry.error(errorMessage)
+            LOG.error { errorMessage }
+            return null
+        }
+    }
+
+    fun rejectHil() {
+        try {
+            codeTransformationSession?.rejectHilAndContinue()
+        } catch (e: Exception) {
+            throw e
```

Comment:
Question: Why don't we log for all exceptions. Close this comment out or add logging.